### PR TITLE
Add e2e tests. Add confirm action and correct action via prompt

### LINKED ISSUE #52: e2e check for v1
Add a basic e2e test that recreates https://github.com/alephmatic/aleph0/issues/8
1. Create a form that allows to post a video.
2. Create a page to see all videos in a table format.
3. Create a page to view videos.
4. Create an upvote component to upvote a video.

### PR DESCRIPTION
**Pull request by agamm**
First I'm trying to add a basic e2e test that would:
1. Create an API route that just returns "Hello aleph0".
2. Then will create a page with a button that when clicked will call that API route.
3. Start next and use playwright to test the whole e2e flow. (TBD)

Problems:
- [ ] (1) API route adds a validation even though it doesn't need it.
- [ ] (2) For some reason it doesn't create a simple page, it has to create a component and calls it ProfileForm.tsx (critical)

Fixes: https://github.com/alephmatic/aleph0/issues/52

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -8,7 +8,8 @@
   "scripts": {
     "watch": "tsup --watch",
     "start": "node dist/index.js",
-    "build": "tsup"
+    "build": "tsup",
+    "test:e2e_hello": "bun tests/e2e-hello.spec.ts"
```
Comment (elie222) [inline package.json]: just make this `test`. you can limit tests that run when you need it with `.only`.

also would recommend jest for testing. look at playwright docs for setting up standard e2e tests. i would not try recreate the wheel here.

File: tests/e2e-hello.spec.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  }
+
+  const options = {
+    projectDir: tempDir,
+    regenerateDescription: true,
+    technology: "next14",
+    model: "gpt-4-1106-preview",
+  } as GenerateOptions;
+
+  generate("create an api at /api/hello that returns 'Hello aleph0'", options);
```
Comment (elie222) [inline tests/e2e-hello.spec.ts]: you need to `await` here i believe.

another way to test via actual cli is using `execa` package. but this probs also fine. but not really full e2e

File: tests/e2e-hello.spec.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    const examplesDir = join(__dirname, "../examples/next/");
+    copySync(examplesDir, tempDir);
+  }
+
+  const options = {
+    projectDir: tempDir,
+    regenerateDescription: true,
+    technology: "next14",
+    model: "gpt-4-1106-preview",
+  } as GenerateOptions;
```
Comment (elie222) [inline tests/e2e-hello.spec.ts]: better way to do this:
don't use `as`.

but:

```ts
const options: GenerateOptions = {
    projectDir: tempDir,
    regenerateDescription: true,
    technology: "next14",
    model: "gpt-4-1106-preview",
  };
```

This type checks it. `as` doesn't. Explained here:
CleanShot 2023-11-28 at 23 20 30@2x

File: package.json (hunk 1 above)
Comment (agamm) [reply package.json]: tried with playwright, not so easy, give `npm run test` a try. @elie222

**Comment by agamm** [comment]
@elie222 
1. `ora` is being funky inside the playwright context.
2. Aleph is not behaving as it should in the test, we may need to fix some stuff - prompting might be.

File: package.json [hunk 4]
```diff
@@ -8,7 +8,8 @@
   "scripts": {
     "watch": "tsup --watch",
     "start": "node dist/index.js",
-    "build": "tsup"
+    "build": "tsup",
+    "test:e2e": "pnpm exec playwright test"
```
Comment (elie222) [inline package.json]: weird you'd need pnpm exec. running `pnpm test:e2e` i would have thought was enough. with `playwright test`

**Comment by agamm** [comment]
@elie222 Is it working now?

**Review by agamm — COMMENTED** [review]
@elie222 Looks cool, would potentially add a reasoning explanation before each action too.

**Pull request merged**